Content.Server/VendingMachines/VendingMachineComponent.cs

```diff
@@ -1,283 +1,246 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using Content.Server.Popups;
 using Content.Server.Power.Components;
 using Content.Server.UserInterface;
 using Content.Server.WireHacking;
-using Content.Shared.Access.Components;
-using Content.Shared.Access.Systems;
 using Content.Shared.Acts;
-using Content.Shared.Interaction;
 using Content.Shared.Sound;
 using Content.Shared.VendingMachines;
 using Robust.Server.GameObjects;
-using Robust.Shared.Audio;
 using Robust.Shared.GameObjects;
 using Robust.Shared.IoC;
-using Robust.Shared.Localization;
-using Robust.Shared.Player;
-using Robust.Shared.Prototypes;
-using Robust.Shared.Random;
 using Robust.Shared.Serialization.Manager.Attributes;
 using Robust.Shared.ViewVariables;
-using static Content.Shared.Wires.SharedWiresComponent;
+using System.Threading;
+using Robust.Shared.Maths;
+using Content.Server.VendingMachines.systems;
 
+using static Content.Shared.Wires.SharedWiresComponent;
+using static Content.Shared.Wires.SharedWiresComponent.WiresAction;
 namespace Content.Server.VendingMachines
 {
     [RegisterComponent]
-    [ComponentReference(typeof(IActivate))]
-    public class VendingMachineComponent : SharedVendingMachineComponent, IActivate, IBreakAct, IWires
+    public class VendingMachineComponent : SharedVendingMachineComponent, IWires, IBreakAct
     {
         [Dependency] private readonly IEntityManager _entMan = default!;
-        [Dependency] private readonly IRobustRandom _random = default!;
-        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [ComponentDependency] private readonly WiresComponent? WiresComponent = null;
 
-        private bool _ejecting;
-        private TimeSpan _animationDuration = TimeSpan.Zero;
+        private VendingMachineSystem? _vendingMachineSystem;
+        public bool Ejecting;
+        public TimeSpan _animationDuration = TimeSpan.Zero;
         [DataField("pack")]
-        private string _packPrototypeId = string.Empty;
-        private string _spriteName = "";
-
-        private bool Powered => !_entMan.TryGetComponent(Owner, out ApcPowerReceiverComponent? receiver) || receiver.Powered;
-        private bool _broken;
-
+        public string PackPrototypeId = string.Empty;
+        public string SpriteName = "";
+        public bool Powered => (PowerPulsed || PowerCut) ? false : !_entMan.TryGetComponent(Owner, out ApcPowerReceiverComponent? receiver) || receiver.Powered;
```

Comment:
Components are not allowed to have calculated properties. Move this logic to `VendingMachineSystem`